include/bout/empty_from.hxx

```diff
@@ -0,0 +1,21 @@
+#ifndef __NEW_EMPTY_FIELD_H__
+#define __NEW_EMPTY_FIELD_H__
+
+#include "field.hxx"
+#include "fieldperp.hxx"
+
+/// Return an empty shell field of some type derived from Field, with metadata
+/// copied but empty data array
+template<typename T>
+inline T emptyFrom(const T& f) {
```

Comment:
That's probably better. The reason for having a separate file was that I wanted to put the general version and the `FieldPerp` specialization in the same place; I couldn't include `fieldperp.hxx` in `field.hxx` because of the circular dependency, so put them both in a new file. In hindsight it would be better to have `emptyFrom` in `field.hxx` and the `FieldPerp` specialization in `fieldperp.hxx`.